storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -137,13 +137,10 @@ public RemoteIndexCache(long maxSize, RemoteStorageManager remoteStorageManager,
     public void resizeCacheSize(long remoteLogIndexFileCacheSize) {
         lock.writeLock().lock();
         try {
-            // When resizing the cache, we always start with an empty cache. There are two main reasons:
-            // 1. Resizing the cache is not a high-frequency operation, and there is no need to fill the data in the old
+            // When resizing the cache, we always start with an empty cache.
+            // Resizing the cache is not a high-frequency operation, and there is no need to fill the data in the old
             // cache to the new cache in time when resizing inside.
-            // 2. Since the eviction of the caffeine cache is cleared asynchronously, it is possible that after the entry
-            // in the old cache is filled in the new cache, the old cache will clear the entry, and the data in the two caches
-            // will be inconsistent.
-            internalCache.invalidateAll();
+            removeAll(internalCache.asMap().keySet());
```

Comment:
OK, we can keep this change, there will be update at this: https://github.com/apache/kafka/pull/14511 . cc @hudeqi

Reply:
Reverted the change